include/rocksdb/options.h

```diff
@@ -847,6 +847,18 @@ struct DBOptions {
   //
   // Dynamically changeable through SetDBOptions() API.
   bool avoid_flush_during_shutdown = false;
+
+  // If enabled it uses two queues for writes, one for the ones with
+  // disable_memtable and one for the ones that also write to memtable. This
+  // allows the memtable writes not to lag behind other writes. It can be used
+  // to optimize MySQL 2PC in which only the commits, which are serial, write to
+  // memtable.
+  bool separate_queue_for_memtable_writes = false;
+
+  // If true WAL is not flushed automatically after each write. Instead it
+  // relies on manual invocation of FlushWAL to write the WAL buffer to its
+  // file.
+  bool manual_wal_flush = false;
```

Comment:
There is an existing use case for only WAL buffering: https://groups.google.com/d/msg/rocksdb/ngNeSSQHfao/_AV_KIf1CAAJ
This feature is going to stay even after we discard concurrent wal writes in future so it does need its own config.

To me, WAL buffer is not a performance config, and it is more of a durability one. Normal users should not be bothered with playing around with them. Advanced users could make use of it only after making changes to their application to call FlushWAL at right times (like myrocks does). So I do not think we should include such options in the tuning complexity issue that we have.